test/Numerics/DGmethods/advection_diffusion/advection_diffusion_model_1dimex_bgmres.jl

```diff
@@ -0,0 +1,406 @@
+    state,
+    aux,
+    x,
+    t,
+) where {n, α, β, μ, δ}
+    ξn = dot(n, x)
+    # ξT = SVector(x) - ξn * n
+    state.ρ = exp(-(ξn - μ - α * t)^2 / (4 * β * (δ + t))) / sqrt(1 + t / δ)
+end
+Dirichlet_data!(P::Pseudo1D, x...) = initial_condition!(P, x...)
+function Neumann_data!(
+    ::Pseudo1D{n, α, β, μ, δ},
+    ∇state,
+    aux,
+    x,
+    t,
+) where {n, α, β, μ, δ}
+    ξn = dot(n, x)
+    ∇state.ρ =
+        -(
+            2n * (ξn - μ - α * t) / (4 * β * (δ + t)) *
+            exp(-(ξn - μ - α * t)^2 / (4 * β * (δ + t))) / sqrt(1 + t / δ)
+        )
+end
+
+function do_output(mpicomm, vtkdir, vtkstep, dg, Q, Qe, model, testname)
+    ## name of the file that this MPI rank will write
+    filename = @sprintf(
+        "%s/%s_mpirank%04d_step%04d",
+        vtkdir,
+        testname,
+        MPI.Comm_rank(mpicomm),
+        vtkstep
+    )
+
+    statenames = flattenednames(vars_state(model, eltype(Q)))
+    exactnames = statenames .* "_exact"
+
+    writevtk(filename, Q, dg, statenames, Qe, exactnames)
+
+    ## Generate the pvtu file for these vtk files
+    if MPI.Comm_rank(mpicomm) == 0
+        ## name of the pvtu file
+        pvtuprefix = @sprintf("%s/%s_step%04d", vtkdir, testname, vtkstep)
+
+        ## name of each of the ranks vtk files
+        prefixes = ntuple(MPI.Comm_size(mpicomm)) do i
+            @sprintf("%s_mpirank%04d_step%04d", testname, i - 1, vtkstep)
+        end
+
+        writepvtu(pvtuprefix, prefixes, (statenames..., exactnames...))
+
+        @info "Done writing VTK: $pvtuprefix"
+    end
+end
+
+
+function run(
+    mpicomm,
+    ArrayType,
+    dim,
+    topl,
+    N,
+    timeend,
+    FT,
+    dt,
+    n,
+    α,
+    β,
+    μ,
+    δ,
+    vtkdir,
+    outputtime,
+    fluxBC,
+)
+
+    grid = DiscontinuousSpectralElementGrid(
+        topl,
+        FloatType = FT,
+        DeviceArray = ArrayType,
+        polynomialorder = N,
+    )
+    model = AdvectionDiffusion{dim, fluxBC}(Pseudo1D{n, α, β, μ, δ}())
+    dg = DGModel(
+        model,
+        grid,
+        Rusanov(),
+        CentralNumericalFluxDiffusive(),
+        CentralNumericalFluxGradient(),
+        direction = EveryDirection(),
+    )
+
+    vdg = DGModel(
+        model,
+        grid,
+        Rusanov(),
+        CentralNumericalFluxDiffusive(),
+        CentralNumericalFluxGradient(),
+        auxstate = dg.auxstate,
+        direction = VerticalDirection(),
+    )
+
+    Q = init_ode_state(dg, FT(0))
+
+    # Setup columnwise GMRES solver (Batched GMRES)
+    topology = grid.topology
+    ngl = N + 1
+    num_states = size(Q)[2]
+    nelem = length(topology.elems)
+    nvertelem = topology.stacksize
+    nhorzelem = div(nelem, nvertelem)
+
+    reshaping_tup = (ngl, ngl, ngl, num_states, nvertelem, nhorzelem)
+    m = reshaping_tup[3] * reshaping_tup[5]
+    n = reshaping_tup[1] * reshaping_tup[2] * reshaping_tup[4] * reshaping_tup[6]
+
+    linearsolver = BatchedGeneralizedMinimalResidual(
```

Comment:
Note for us to remember: Since all the reshaping information can be obtained from the grid topology, it might be worth adding a constructor for `BatchedGeneralizedMinimalResidual` to take a DG model in this PR. From there, the constructor can take care of generating the reshape tuples. It will simplify how a user instantiates this iterative method.

Reply:
@sandreza @leios I pushed the driver and updated the advection test `advection_diffusion_model_1dimex-bgmres.jl`. It seems to be working. Let me know if you are both happy with it.

Reply:
I like it! I've tested it on our cluster as well, and all the tests passed